### DISCUSSION TITLE
Replacing a Graphite query

### DISCUSSION THREAD

**Discussion by JaskaJ78** [NetApp/harvest#917] [General]
Hello,

Does anyone know how I could replace the Graphite's maxSeries() function in Prometheus?
For example, for billing purposes we pick up the highest number between LUN size and actual volume used:
maxSeries(netapp.capacity.{zone}.{cluster}.svm.{svm}.vol.{vol}.actual_volume_size, netapp.capacity.{zone}.{cluster}.svm.{svm}.vol.{vol}.lun.{lun}.lun_size)
     
To clarify, just need to replace the maxSeries() function itself. The other parameters I have.

**Comment by cgrinds** [comment]
hi @JaskaJ78 looks like this has been requested of Prometheus in the past https://github.com/prometheus/prometheus/issues/6690

Would something like this work for your usecase?

hat-tip to @rahulguptajss for the syntax

```
lun_size{node="F2240-127-26",volume="Vol_BRLUN1"} > ignoring (aggr,lun,svm,style)
volume_size{node="F2240-127-26",volume="Vol_BRLUN1"} or
volume_size{node="F2240-127-26",volume="Vol_BRLUN1"}
```

**Reply by JaskaJ78** [reply]
Hi, 

Thanks for the tip.
If I use the 'or' at the end, I get two values so I went with this syntax:
lun_size{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"} >ignoring (aggr,lun,svm,style)  volume_space_physical_used{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"} 

I still need to test it with a volume which is actually larger than the LUN size.

**Comment by JaskaJ78** [comment]
Hi @cgrinds 

This solution doesn't seem to fully replace the maxSeries() function.
For example, these two queries return different results:
#1
lun_size_used{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"}>ignoring (aggr,lun,svm,style) volume_size_total{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"}

#2
volume_size_total{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"}>ignoring (aggr,lun,svm,style)lun_size_used{datacenter="$Datacenter",cluster="$Cluster",svm=~"$SVM",volume=~"$Volume"}

First function returns 'No data' and the second function the volume size since it is greater than the LUN size.
I require something that compares the two numbers and returns the higher number regardless of it's position in the function.

**Reply by cgrinds** [reply]
Hi @JaskaJ78 if Prometheus can't be coerced into service, I wonder if Grafana has anything that would help? It may not since it delegates most of that functionality to the datasource.

Few other ideas:
- create a Harvest plugin that does the max calculation at collection time and exports a new metric to Prometheus
- use a [Telegraf](https://github.com/influxdata/telegraf) or a 3rd party plugin to do the max aggregation for you and export a new metric. Maybe this [one](https://github.com/influxdata/telegraf/blob/release-1.20/plugins/aggregators/basicstats/README.md) would work?
